doc/api-extensions.md

```diff
@@ -219,12 +219,12 @@ Introduces the ability to rename a volume group by setting "storage.lvm.vg\_name
 Introduces the ability to rename a thinpool name by setting "storage.thinpool\_name".
 
 ## network\_vlan
-This adds a new "vlan" property to "macvlan" network devices.
+This adds a new "vlan" property to "macvlan" and "physical" network devices.
 
 When set, this will instruct LXD to attach to the specified VLAN. LXD
 will look for an existing interface for that VLAN on the host. If one
 can't be found it will create one itself and then use that as the
-macvlan parent.
+macvlan parent if "macvlan" nictype is selected or as it if "physical" nictype is selected.
```

Comment:
Hi Stéphane,

Sorry for this late answer but I was busy and I needed some time to think about your answer and to talk with some colleagues.

I still disagree with the need to create a new property. An API is a way for entities to understand each other using a grammar and a syntax. An API should not reflect the ability to do something. For instance, if I ask you to swim, you will understand what I want you to do even if you can't swim, then you will answer me "I can't".

I plan to add "Ethernet over GRE" connectivity to LXD in a future patch... would I need then to add a new "gre_network_vlan" property for exactly the same meaning?

Furthermore, the fact to have "network_vlan" API extension available should not be considered by the client as a proof the fuction is actually there. There are undreds of reasons the function could not be used (kernel module not available, maximum number of interface reached...) so the client has to check the answer for errors.

Best regards
Vincent